opennlp-tools/src/test/java/opennlp/tools/ml/maxent/GISTestIndexing.java

```diff
@@ -0,0 +1,91 @@
+package opennlp.tools.ml.maxent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import opennlp.tools.ml.model.Event;
+import opennlp.tools.util.ObjectStream;
+import opennlp.tools.util.ObjectStreamUtils;
+
+public class GISTestIndexing {
+
+  static String[][] cntx = new String[][]{
+    {"dog","cat","mouse"},
+    {"text", "print", "mouse"},
+    {"dog", "pig", "cat", "mouse"}
+  };
+  static String[] outputs = new String[]{"A","B","A"};
+
+  /*
+   * Test the GIS.trainModel(ObjectStream<Event> eventStream) method
+   */
+  @Test
+  public void testGISTrainSignature1() throws Exception {
+    List<Event> events = new ArrayList<>();
+    for (int i = 0; i < cntx.length; i++) {
+      events.add(new Event(outputs[i], cntx[i]));
+    }
+    ObjectStream<Event> eventStream = ObjectStreamUtils.createObjectStream(events);
+    Assert.assertNotNull(GIS.trainModel(eventStream));
+    eventStream.close();
+  }
+
+  /*
+   * Test the GIS.trainModel(ObjectStream<Event> eventStream,boolean smoothing) method
+   */
+  @Test
+  public void testGISTrainSignature2() throws Exception {
+    List<Event> events = new ArrayList<>();
+    for (int i = 0; i < cntx.length; i++) {
+      events.add(new Event(outputs[i], cntx[i]));
+    }
+    ObjectStream<Event> eventStream = ObjectStreamUtils.createObjectStream(events);
+    Assert.assertNotNull(GIS.trainModel(eventStream,true));
+    eventStream.close();
+  }
+  
+  /*
+   * Test the GIS.trainModel(ObjectStream<Event> eventStream, int iterations, int cutoff) method
+   */
+  @Test
+  public void testGISTrainSignature3() throws Exception {
+    List<Event> events = new ArrayList<>();
+    for (int i = 0;i < cntx.length;i++) {
+      events.add(new Event(outputs[i], cntx[i]));
+    }
+    ObjectStream<Event> eventStream = ObjectStreamUtils.createObjectStream(events);
+    Assert.assertNotNull(GIS.trainModel(eventStream,10,1));
+    eventStream.close();
+  }
+ 
+  /*
+   * Test the GIS.trainModel(ObjectStream<Event> eventStream, int iterations, int cutoff, double sigma) method
+   */
+  @Test
+  public void testGISTrainSignature4() throws Exception {
+    List<Event> events = new ArrayList<>();
+    for (int i = 0;i < cntx.length;i++) {
+      events.add(new Event(outputs[i], cntx[i]));
+    }
+    ObjectStream<Event> eventStream = ObjectStreamUtils.createObjectStream(events);
+    Assert.assertNotNull(GIS.trainModel(eventStream,10,1,0.01));
+    eventStream.close();
+  }
+  
+  /*
+   * Test the GIS.trainModel((ObjectStream<Event> eventStream, int iterations, int cutoff, boolean smoothing, boolean printMessagesWhileTraining)) method
```

Comment:
Lines shouldn't be longer than 100 chars. Sorry, we don't yet have a checkstyle for that.